tests/cupy_tests/lib_tests/test_polynomial.py

```diff
@@ -403,3 +425,92 @@ def test_polyadd_diff_types_poly1d(self, xp, dtype1, dtype2):
             out = xp.polyadd(a, b)
         assert out.variable == 'x'
         return out
+
+
+@testing.gpu
+class TestRoots(unittest.TestCase):
+
+    @testing.for_signed_dtypes()
+    @testing.numpy_cupy_allclose(rtol=1e-6)
+    @_wraps_root_test
+    def test_roots1(self, xp, dtype):
+        a = xp.array([2, -1, -2], dtype)
+        return xp.roots(a)
+
+    @testing.for_signed_dtypes()
+    @testing.numpy_cupy_allclose(rtol=1e-6)
+    @_wraps_root_test
+    def test_roots2(self, xp, dtype):
+        a = xp.array([-4, 10, 4], dtype)
+        return xp.roots(a)
+
+    @testing.for_complex_dtypes()
+    @testing.numpy_cupy_allclose(rtol=1e-6)
+    @_wraps_root_test
+    def test_roots_complex(self, xp, dtype):
+        a = xp.array([3j, 1.5j, -3j], dtype)
+        return xp.roots(a)
+
+    @testing.for_all_dtypes(no_float16=True, no_bool=True)
+    @testing.numpy_cupy_array_equal()
+    def test_roots_two_sized1(self, xp, dtype):
+        a = xp.array([5, 10], dtype)
+        return xp.roots(a)
+
+    @testing.for_all_dtypes(no_bool=True)
```

Comment:
bool tests are being skipped because of as_series. I think trying a workaround to support them isn't necessary since roots should be numerical values ?

Reply:
How about raising `NotImplementedError` for boolean inputs?